addons/calendar/security/calendar_security.xml

```diff
@@ -1,19 +1,47 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo>
-    <data noupdate="1">
+    <data noupdate="0">
 
-        <record id="calendar_event_rule_my" model="ir.rule">
-            <field name="name">Own events</field>
+        <record id="calendar_event_admin_write" model="ir.rule">
+            <field name="name">Calendar Event: Admin can write all events</field>
             <field ref="model_calendar_event" name="model_id"/>
-            <field name="domain_force">[('partner_ids', 'in', user.partner_id.id)]</field>
-            <field name="groups" eval="[(4, ref('base.group_portal'))]"/>
+            <field name="domain_force">[('privacy', '!=', 'private')]</field>
+            <field name="perm_unlink" eval="1"/>
+            <field name="perm_write" eval="1"/>
+            <field name="perm_read" eval="1"/>
+            <field name="perm_create" eval="1"/>
+            <field name="groups" eval="[(4, ref('base.group_partner_manager'))]"/>
+        </record>
+
+        <record id="calendar_event_own_event_write" model="ir.rule">
+            <field name="name">Calendar Event: write own events</field>
+            <field ref="model_calendar_event" name="model_id"/>
+            <field name="domain_force">['|', ('partner_ids.user_ids', 'in', user.id), ('user_id.id', '=', user.id)]</field>
+            <field name="perm_unlink" eval="1"/>
+            <field name="perm_write" eval="1"/>
+            <field name="perm_read" eval="1"/>
+            <field name="perm_create" eval="1"/>
+            <field name="groups" eval="[(4, ref('base.group_user'))]"/>
         </record>
 
         <record id="calendar_event_rule_employee" model="ir.rule">
             <field ref="model_calendar_event" name="model_id"/>
-            <field name="name">All Calendar Event for employees</field>
+            <field name="name">Calendar Event: Internal user can read all events</field>
             <field name="domain_force">[(1, '=', 1)]</field>
-            <field eval="[(4,ref('base.group_user'))]" name="groups"/>
+            <field name="perm_unlink" eval="0"/>
+            <field name="perm_write" eval="0"/>
+            <field name="perm_read" eval="1"/>
+            <field name="perm_create" eval="1"/>
+            <field name="groups" eval="[(4,ref('base.group_user'))]"/>
+        </record>
+
+        <!-- ARJ TODO: Do we still need portal rules ? -->
+
+        <record id="calendar_event_rule_my" model="ir.rule">
+            <field name="name">Own events</field>
+            <field ref="model_calendar_event" name="model_id"/>
+            <field name="domain_force">[('partner_ids', 'in', user.partner_id.id)]</field>
+            <field name="groups" eval="[(4, ref('base.group_portal'))]"/>
         </record>
 
         <record id="calendar_attendee_rule_my" model="ir.rule">
```

Comment:
Doesn't keeping that rule allow everybody to read every attendee ? + Modify all the attendees ?